packages/menu/src/MenuDescendant/MenuDescendant.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { ComponentPropsWithRef, ReactElement } from 'react';
+import PropTypes from 'prop-types';
+
+import { useDescendant } from '@leafygreen-ui/descendants';
+import { cx } from '@leafygreen-ui/emotion';
+
+import { MenuDescendantsContext, useMenuContext } from '../MenuContext';
+
+import { getMenuDescendantWrapperStyles } from './MenuDescendant.styles';
+
+/**
+ * Renders arbitrary focusable components as descendants in a menu context.
+ *
+ * By wrapping a single focusable component in `MenuDescendant`,
+ * it will be registered as a menu item descendant,
+ * and will be focusable using standard menu arrow key interactions.
+ *
+ * Note: the single child of `MenuDescendant` must _itself_ be focusable.
+ * Wrapping a focusable element (e.g. `input` in a `div`)
+ * will not enable the menu descendant functionality.
+ */
+export const MenuDescendant = React.forwardRef<
+  HTMLElement,
+  ComponentPropsWithRef<'div'>
+>(({ children, className, ...rest }, fwdRef) => {
+  const { theme } = useMenuContext();
+  const { ref, id } = useDescendant(MenuDescendantsContext, fwdRef, {});
+  const child = React.Children.only(children) as ReactElement;
+
+  return child ? (
+    <div
+      id={id}
+      className={cx(getMenuDescendantWrapperStyles(theme))}
+      {...rest}
+    >
+      {React.cloneElement(child, {
+        ref,
```

Comment:
does the id and ref need to be on the same element for the descendant pattern to work correctly or no? also do we want to pass an aria-role while we're at it?

Reply:
- `id` doesn't matter for tracking descendants, only the ref. The `id` is just used internally as an identifier. 
- The role would change depending on the element, no? If we're baking a lot of things in, why not keep it as an `InputOption`?

Reply:
InputOption enforces way too much style. This is just about making something usable/accessible by Menu specifically without enforcing any stylistic rules.